backend/src/data/repositories/slc-function/slc-function.repository.ts

```diff
@@ -0,0 +1,82 @@
+import { SLCFunction as SLCFunctionM } from '~/data/models/models';
+import { FunctionTableField } from '~/data/models/slc-function/slc-function-table-field.enum';
+import { SLCFunction as SLCFunctionEntity } from '~/services/slc-function/slc-function.entity';
+
+type Constructor = {
+  SLCFunctionModel: typeof SLCFunctionM;
+};
+
+class SLCFunction {
+  #SLCFunctionModel: typeof SLCFunctionM;
+
+  constructor({ SLCFunctionModel }: Constructor) {
+    this.#SLCFunctionModel = SLCFunctionModel;
+  }
+
+  async getByName({
+    workersIds,
+    name,
+  }: {
+    workersIds: string[];
+    name: string;
+  }): Promise<SLCFunctionEntity | null> {
+    const slcFunction = await this.#SLCFunctionModel
+      .query()
+      .select()
+      .whereIn(FunctionTableField.CREATED_BY, workersIds)
+      .andWhere({ name })
+      .first();
```

Comment:
```suggestion
  async getByNameAndWorkerId({
    workerId,
    name,
  }: {
    workerId: string;
    name: string;
  }): Promise<SLCFunctionEntity | null> {
    const slcFunction = await this.#SLCFunctionModel
      .query()
      .select()
      .where(FunctionTableField.CREATED_BY, workerId)
      .andWhere({ name })
      .first();
```

Reply:
In the criteria of task #21, the function name must be unique for the tenant